servers/rendering/shader_language.cpp

```diff
@@ -6589,7 +6589,7 @@ Error ShaderLanguage::_parse_block(BlockNode *p_block, const FunctionInfo &p_fun
 									struct_name2 = tk.text;
 								} else {
 									if (!is_token_variable_datatype(tk.type)) {
-										_set_error("Invalid data type for array");
+										_set_error(RTR("The invalid data type for the array."));
```

Comment:
```suggestion
										_set_error(RTR("Invalid data type for the array."));
```
Basically when a sentence starts with "The", it calls for a "full" sentence with subject, verb, object, so this would need to be "The data type for the array is invalid." but in tech language "Invalid data type for the array." is more usual as it's straight to the point.